### PR TITLE
Disappearing item trade fix

### PR DESCRIPTION
**Pull request by nekiro**
This is alternative version of fix provided by @mackerel225 in #2966
also now player and partner variable are guaranteed to be the real trade requester and trade partner, does not matter who accepted the trade last.

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -2731,10 +2731,17 @@ void Game::playerAcceptTrade(uint32_t playerId)
 			if (tradePartnerRet == RETURNVALUE_NOERROR && playerRet == RETURNVALUE_NOERROR) {
 				tradePartnerRet = internalMoveItem(playerTradeItem->getParent(), tradePartner, INDEX_WHEREEVER, playerTradeItem, playerTradeItem->getItemCount(), nullptr, FLAG_IGNOREAUTOSTACK, nullptr, partnerTradeItem);
 				if (tradePartnerRet == RETURNVALUE_NOERROR) {
-					internalMoveItem(partnerTradeItem->getParent(), player, INDEX_WHEREEVER, partnerTradeItem, partnerTradeItem->getItemCount(), nullptr, FLAG_IGNOREAUTOSTACK);
-					playerTradeItem->onTradeEvent(ON_TRADE_TRANSFER, tradePartner);
-					partnerTradeItem->onTradeEvent(ON_TRADE_TRANSFER, player);
-					isSuccess = true;
+					// check again if we can move this item to player
+					playerRet = internalAddItem(player, partnerTradeItem, INDEX_WHEREEVER, 0, true);
```
Comment (mackerel225) [inline src/game.cpp:2735]: It's almost functional, there is still bug which depends on the last person who pressed 'Accept' button. This means, you also need to check again for **tradePartnerRet**

Comment (nekiro) [reply src/game.cpp:2735]: at this point player item is already in partner inventory, so all that is left is to check if partner item can be moved to player. Could you elaborate futher cause maybe I dont understand what you mean

Comment (mackerel225) [reply src/game.cpp:2735]: When I was working on this function yesterday I was trying different solutions, including something similar to yours. There was some sort of inconsistency with who is **player** and **tradePartner**. After several minutes of testing, it became clear that the **player** and **tradePartner** change based on who finalised the trade. By 'finalise', we mean the player who pressed 'Accept' button last. This comes down to what is written in the code right now.

Consider these two scenarios:

1.
Player 1 initialised trade with Player 2
Player 1 finalised the trade by pressing Accept button last.
This results in bypassing your code.

2.
Player 1 initialised trade with Player 2
Player 2 finalised the trade by pressing Accept button last.
This results in correct execution of your code.

**Comment by nekiro** [comment]
@mackerel225 updated the code, care to check it again?

**Comment by nekiro** [comment]
I do? Tell me the steps to reproduce any kind of bug with current code, because I tested it and seem to work just fine. No vanishing items.

**Comment by mackerel225** [comment]
> 
> 
> I do? Tell me the steps to reproduce any kind of bug with current code, because I tested it and seem to work just fine. No vanishing items.

Here is a short video:

Imgur Image

**Pull request closed without merging**